lib/public/models/OverviewModel.js

```diff
@@ -20,3 +26,120 @@
 /**
  * @property {PaginationModel} OverviewModel#pagination pagination model of the overview
  */
+
+/**
+ * Base model for an overview page
+ *
+ * @template T the type of data displayed in the overview page
+ */
+export class OverviewPageModel extends Observable {
+    /**
+     * Constructor
+     */
+    constructor() {
+        super();
+
+        this._pagination = new PaginationModel();
+        this._pagination.observe(() => this.load());
+        this._pagination.itemsPerPageSelector$.observe(() => this.notify());
+
+        const dataSourceObservable = ObservableData.builder().initialValue(RemoteData.loading()).build();
```

Comment:
I think it's good later on in the code to know that this observable is mapped to the data of the datasource. No?

Reply:
yes, I meant just to name it like `dataSourceObservableData`, but I am fine with current one

Reply:
Anyway, is there some advantage of having intermediate `ObservableData` ?
In Constructor, there might be:
```
        this._observableItems = ObservableData.builder()
            .initialValue(RemoteData.notAsked())
            .map((remoteData) => this.parseApiRemoteData(remoteData))
            .build();
        this._observableItems.bubbleTo(this);

        this._dataSource = new PaginatedRemoteDataSource();
        this._dataSource.pipe(this._observableItems);
``` 
and in parsing method:
```
  return remoteData.match({
            NotAsked: () => RemoteData.notAsked(),
            Failure: (error) => RemoteData.failure(error),
            Loading: () => keepExisting
                ? this._observableItems.getCurrent()
                : RemoteData.loading(),
            Success: ({ items, totalCount }) => {
                const concatenateWith = keepExisting ? this.items.match({
                    Success: (payload) => payload,
                    Other: () => [],
                }) : [];
                this._pagination.itemsCount = totalCount;
                return RemoteData.success([...concatenateWith, ...this.processItems(items)]);
            },
        });
```

Reply:
Because for me it made more sense. I think it weird in the parse function to set the value of pagination & to return the data of the page. I prefer to set both, hence the separation